docs/faqs.md

```diff
@@ -0,0 +1,153 @@
+
+This is something we wish to avoid — Jeff is *not* a standard we need all stakeholders to adopt, but
+an exchange format localized to compilers, designed to
+be easily extensible. Additional information for program representation and compilation can be added
+directly, and will be taken into account by software frameworks that understand the direction.
+Further, development will happen out in the open, on GitHub.
+
+## Why not format X?
+
+### Why not OpenQASM2?
+
+[OpenQASM2](https://arxiv.org/abs/1707.03429) has become the de-facto interchange format in quantum software, but the biggest drawback
+has been the lack of program structure — something that we *need* to be a first-class citizen in a
+compiler exchange format.
+
+An exchange format that makes it easy to incorporate structure will also allow the field to consider
+compilation of structured programs by default.
+
+### Why not OpenQASM3?
+
+[OpenQASM3](https://arxiv.org/abs/2104.14722) brings program structure to OpenQASM2, however has not yet seen adoption or tooling
+maturity in the ecosystem to the extent as OpenQASM2. In addition, OpenQASM3 maintains a focus on
+human readability (requiring string generation and parsing), and is not designed first and foremost for quantum optimization.
+
+Moreover, OpenQASM3 has a few quirks that limit its general usability as a compiler exchange
+format:
+
+- Lack of extensibility without evolving the specification (a `#pragma` directive is provided, but
+  does not easily allow for including general structured information).
+
+- You can only index into arrays using compile-time constants, not dynamic values that may only been
+  known at runtime.
+
+- Registers (TODO).
+
+### Why not QIR?
+
+[QIR](https://www.qir-alliance.org/) was originally designed as a low-level, hardware-facing IR for optimization, for example using
+the QIR Adapter Tool (QAT). However, QIR today is used less for quantum optimization as it is quite
+low-level (missing a lot of the higher-level abstractions we would like to preserve during
+compilation), and is predominantly used as a low-level interchange format between software
+frameworks and hardware providers.
+
+This can be seen by looking at the various software compilers built on top on the QIR stack such as
+[CUDA-Q](https://github.com/nvidia/cuda-quantum) and [Catalyst](https://github.com/pennylaneai/catalyst) — rather than compiling QIR programs, they instead
+optimize MLIR, a higher-level intermediate representation.
+
+Furthermore, relying on QIR as a compilation exchange format introduces a big barrier to entry due
+to it depending on the LLVM toolchain — a massive software project that has non-trivial
+compatibility concerns, installation, and learning overhead.
+
+### Why not MLIR?
+
+[MLIR (Multi-Level Intermediate Representation)](https://en.wikipedia.org/wiki/MLIR_(software)) is a framework that exists on top of QIR/LLVM, and
```

Comment:
MLIR doesn't exist on top of QIR, you could argue it exists on top of LLVM (a lof of inspiration is taken from llvm, mlir uses a lot of core support libraries from llvm, and there is built-in lowering from standard mlir dialects to llvm ir), although even that is not necessarily a conceptual requirement for mlir (it can be used without going to llvm ir, or implemented without llvm libraries like in xdsl).

Reply:
```suggestion
[MLIR (Multi-Level Intermediate Representation)](https://en.wikipedia.org/wiki/MLIR_(software)) is a framework that developed by the LLVM project, and
```